client/containers/DashboardSubmissions/VerdictDialog.tsx

```diff
@@ -0,0 +1,154 @@
+	return (
+		<>
+			<div className={Classes.DIALOG_BODY}>
+				<p>Would you like to {props.actionTitle} this submission?</p>
+				<p className="email-text-header">
+					<Icon icon="manually-entered-data" iconSize={12} />
+					{'  '}Email to Authors
+				</p>
+				<WorkflowTextEditor
+					initialContent={customEmailText}
+					onContent={setCustomEmailText}
+					placeholder="Specify message to pub author(s)."
+				/>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Checkbox
+						className="email-toggle"
+						checked={shouldSendEmail}
+						disabled={props.isHandlingSubmission}
+						onChange={(e) => {
+							setShouldSendEmail((e.target as HTMLInputElement).checked);
+						}}
+						label="Notify authors by email"
+					/>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Cancel
+					</Button>
+					<Button
+						onClick={() => props.handleSubmission(customEmailText, shouldSendEmail)}
+						loading={props.isHandlingSubmission}
+						intent="primary"
+					>
+						Email & {props.actionTitle}
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+type PostSubmitBodyProps = {
+	completedName: string;
+	onClose: () => unknown;
+	isHandlingSubmission: boolean;
+};
+const PostSubmitBody = (props: PostSubmitBodyProps) => {
+	return (
+		<>
+			<div className={Classes.DIALOG_BODY}>
+				<Callout intent="success" title={`Submission ${props.completedName}!`}>
+					You successfully {props.completedName} the submission!
+				</Callout>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Close
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+const VerdictDialog = (props: Props) => {
+	const [isHandlingSubmission, setIsHandlingSubmission] = useState(false);
+	const [updatedSubmission, setUpdatedSubmission] = useState(null);
+	const [submissionError, setSubmissionError] = useState(null);
+	const handleSubmission = (customEmailText: DocJson, shouldSendEmail: boolean) => {
+		setIsHandlingSubmission(true);
+		apiFetch('/api/submissions', {
+			method: props.apiMethod,
+			body: JSON.stringify({
+				id: props.pub.submission.id,
+				skipEmail: !shouldSendEmail,
+				...(customEmailText && { customEmailText }),
+				...(props.status && { status: props.status }),
+			}),
+		})
+			.then((submissionRes) => {
+				setUpdatedSubmission(submissionRes);
+				setIsHandlingSubmission(false);
+			})
+			.then(() => props.onJudgePub(props.pub.id, props.status))
+			.catch((err) => {
+				setSubmissionError(err);
+				setIsHandlingSubmission(false);
+			});
+	};
+	return (
+		<Dialog
+			lazy={true}
+			title={props.actionTitle}
+			className="verdict-dialog"
+			isOpen={props.isOpen}
+			onClose={props.onClose}
+		>
+			{submissionError ? (
```

Comment:
Would you consider a `renderInner()` function with a set of `if` statements? The nested ternaries are a little hard to follow here.

Reply:
I am reluctant to break from chained ternary, but I think it's just a matter of taste. Using a `renderInner()` would mean either (a) passing extraneous props into it only to discard them if the necessary component doesn't need them or (b) priming a few components somewhere inside the `VerdictDialog` component before the return value. Those each smell funkier to me.

Here's what converted me a year or so ago:
https://medium.com/javascript-scene/nested-ternaries-are-great-361bddd0f340

Overall, I'm still on the lookout for the best approach to multiple-choice conditional rendering.

Reply:
Fine by me if you feel strongly about it, though I think the readability argument breaks down when the expressions don't mostly fit onto a single line like this:

```ts
const value =  (!conditionA)
    ? valueC
    : (conditionB)
    ? valueA
    : valueB;
```


 Neither (a) or (b) is necessarily what I'm imagining though. This is a good example of what I mean — the inner React elements are being constructed within the `return` statement, just in an inner function definition.

https://github.com/pubpub/pubpub/blob/a8b1c090a8c523c853d31af520657b630d510b76/client/components/PubShareDialog/PubShareDialog.tsx#L159

Also on (b) — at some point in the last two years @3mcd kind of convinced me that defining React nodes outside of a `return` statement is a totally fine thing to do:

https://github.com/pubpub/pubpub/blob/2965db0ed81ce2f24dd41b59f02cfc0521796edc/client/utils/activity/renderers/itemRenderer.tsx#L28

Just so you know where I am on the carousel of JS code fashions.